tests/unit/anchore_engine/services/policy_engine/engine/vulns/test_providers.py

```diff
@@ -0,0 +1,147 @@
+import json
+import pytest
+
+from anchore_engine.clients.grype_wrapper import (
+    GrypeVulnerability,
+    GrypeVulnerabilityMetadata,
+)
+from anchore_engine.services.policy_engine.engine.vulns.providers import GrypeProvider
+from unittest import TestCase
+
+
+def mock_raw_grype_vulnerability(
```

Comment:
when mock objects require this much state, it hints at an implementation problem. This no longer looks like a unit test since it needs so many things to be set